contracts/staking/StakedSYN.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+interface IMinter {
+    function stakingMint(uint256 lastMint) external returns(uint256);
+}
+
+contract StakedSYN is Ownable, ERC20("Staked Synapse", "sSYN") {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    IERC20 public synapse;
+    IMinter public STAKING_MINTER;
+    // Tracks active delegated SYN backing sSYN
+    uint256 internal activeSYN;
+    // Last timestamp of SYN distributed to contract
+    uint256 internal lastSynMint;
+    
+    // Tracks when SYN can be unstaked, along with amounts at time of unstake
+    mapping(address => uint256) public undelegateTimestamps;
+    mapping(address => uint256) public undelegateUnderlyingAmounts;
+
+    // Define the Synapse token contract
+    constructor(IERC20 _synapse, uint256 _stakingStart) public {
+        synapse = _synapse;
+        lastSynMint = _stakingStart;
+    } 
+
+    /*** RESTRICTED FUNCTIONS ***/
+
+    function setStakingMinter(address _minter) external onlyOwner {
+        STAKING_MINTER = IMinter(_minter);
+    }
+
+    /*** VIEW FUNCTIONS ***/
+
+    function undelegatedSynapse(address user) external view returns(uint256 amount, uint256 timestamp) {
+        return (undelegateUnderlyingAmounts[user], undelegateTimestamps[user]);
+    }
+
+    function _getUnderlyingSynapseAmount(uint256 _amount) internal view returns(uint256) {
+        uint256 totalStaked = totalSupply();
+        return totalStaked > 0 ? _amount.mul(activeSYN).div(totalStaked) : 0;
+    }
+
+    function underlyingBalanceOf(address user) external view returns(uint256) {
+        return _getUnderlyingSynapseAmount(balanceOf(user)) + undelegateUnderlyingAmounts[user];
+    }
+
+    function distribute() public {
+        if (totalSupply() != 0) {
+            uint256 lastMint = lastSynMint;
+            if (block.timestamp >= lastMint.add(1 hours)) {
+                lastSynMint = block.timestamp;
+                uint256 mintAmount = STAKING_MINTER.stakingMint(lastMint);
+                activeSYN = activeSYN.add(mintAmount);
+            }
+        }
+    }
+
+    // Enter the bar. Give some SYN. Get some sSYN.
+    // Locks Synapse and mints sSYN
+    function stake(uint256 _amount) external {
+        // Catch up on 
+        distribute();
+
+        uint256 totalStaked = totalSupply();
+        // If no sSYN exists, mint it 1:1 to the amount put in
+        if (totalStaked == 0 || activeSYN == 0) {
```

Comment:
Can be simplified to just `if (activeSYN == 0)`, since invariant is held: `totalStaked = activeSYN * k`, where `k` grows over time.